ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -166,37 +176,20 @@ const NetworksForm = ({
     setChainId,
     setTicker,
     setBlockExplorerUrl,
-    setErrors,
+    setRpcUrlError,
+    setChainIdError,
+    setBlockExplorerUrlError,
     dispatch,
   ]);
 
-  const setErrorTo = (errorKey, errorVal) => {
-    setErrors({ ...errors, [errorKey]: errorVal });
-  };
-
-  const setErrorEmpty = (errorKey) => {
-    setErrors({
-      ...errors,
-      [errorKey]: {
-        msg: '',
-        key: '',
-      },
-    });
-  };
-
-  const hasError = (errorKey, errorKeyVal) => {
-    return errors[errorKey]?.key === errorKeyVal;
-  };
-
   const hasErrors = () => {
-    return Object.keys(errors).some((key) => {
-      const error = errors[key];
-      // Do not factor in duplicate chain id error for submission disabling
-      if (key === 'chainId' && error.key === 'chainIdExistsErrorMsg') {
-        return false;
-      }
-      return error.key && error.msg;
-    });
+    // Do not factor in duplicate chain id error for submission disabling
+    return (
+      (Boolean(chainIdError?.key) &&
+        chainIdError.key === 'chainIdExistsErrorMsg') ||
```

Comment:
```suggestion
        chainIdError.key !== 'chainIdExistsErrorMsg') ||
```
To confirm, we want to not include the duplicate chain id errors here